modules/modules_test.go

```diff
@@ -414,9 +415,12 @@ func TestModuleService_InterruptedSlowStartup(t *testing.T) {
 
 	<-subserviceStarted
 
-	// If moduleService can handle StartingStates, then it should call Stop on the subservice.
-	require.NoError(t, services.StopAndAwaitTerminated(context.Background(), moduleSvc))
+	// We get context.Canceled from moduleService.start.
+	require.ErrorIs(t, services.StopAndAwaitTerminated(context.Background(), moduleSvc), context.Canceled)
```

Comment:
This is a bit confusing IMO that we get a context cancelation from an unrelated context. I think wrapping the context cancellation will be clearer. I made another suggestion